pkg/controllers/jobset_controller.go

```diff
@@ -612,11 +612,15 @@ func constructJob(js *jobset.JobSet, rjob *jobset.ReplicatedJob, jobIdx int) (*b
 	}
 
 	// If this job should be exclusive per topology, configure the scheduling constraints accordingly.
-	if _, ok := js.Annotations[jobset.ExclusiveKey]; ok {
+	_, jsExclusivePlacement := js.Annotations[jobset.ExclusiveKey]
+	_, rjobExclusivePlacement := rjob.Template.Annotations[jobset.ExclusiveKey]
+	if jsExclusivePlacement || rjobExclusivePlacement {
```

Comment:
actually here, we just need to check `job.Annotations[jobset.ExclusiveKey]`, and below the same, use `job.Annotations[jobset.NodeSelectorStrategyKey]` since the job object was annotated above, right?

Reply:
Ah good point, that's simpler - done.